src/commands.ts

```diff
@@ -1146,28 +1154,42 @@ export const insertTableFootnote = (
   const insertedAt = state.selection.to
 
   const inlineFootnotes = findChildrenByType(node, schema.nodes.inline_footnote)
-  const labels = inlineFootnotes.map(
-    (nodeWithPos) => nodeWithPos.node.attrs.contents
+  const footnoteIndex = inlineFootnotes.findIndex(
+    ({ pos }) => position + pos >= insertedAt
```

Comment:
> Can you explain this?

this will get the index of the newly inserted table footnote

> To make this clear and easier to read, I would suggest having a helper function in footnotes-utils called getNewFootnoteLabel. Also, I was expecting something like InlineFootnotes.filter((n) => n[1] <= insertedAt).length + 1

we don't need to filter the array, we just need the index of the new footnote based on the inline footnote elements in the table

Reply:
> But I meant, can you explain your approach for doing this?

just mapping the inline footnote position to the table footnote element, by using index.

> The filter is used to discard the inline-footnotes placed after the insertion position. After applying the filter, the length of the remaining array plus one represents the index (label) of the new inline-footnote. This is generally the idea behind it. Does that make sense to you?

hmm it's the same as `findIndex`,  updated to use the filter. and plus one will not give us the right position

Reply:
It should be `position + pos <= insertedAt` not `>=`.

inlineFootnotes.filter(({ pos }) => position + pos <= insertedAt).length + 1

Reply:
> inlineFootnotes.filter(({ pos }) => position + pos <= insertedAt).length + 1

that will not work!!

Reply:
updated to your suggestion